### PR TITLE
i262: Preserve text fields when navigating between sign in & create account

### LINKED ISSUE #262: preserve text fields when navigate from sign in screen to sign up
#### Problem
On navigate between sign in/sign up screens, the text fields are being reset. This means the user will have to retype anything from the previous screen and could result in the loss of some golden UX points.

#### Solution
remove reseting text fields (on navigate between sign in/up)

- [ ] CreateAccountLink (onTap)
    - [ ] redux state
    - [ ] local form state
- [ ] SignInLink(onTap)
    - [ ] redux state
    - [ ] local form state

### PR DESCRIPTION
**Pull request by nickmeinhold**
Fixes #262

### CODE REVIEW

File: lib/widgets/auth/email_auth_options_page/links/sign_in_link.dart [hunk 1]
```diff
@@ -14,13 +13,8 @@ class SignInLink extends StatelessWidget {
                 decoration: TextDecoration.underline,
                 color: Colors.blue)),
         onTap: () {
-          Form.of(context).reset();
           context.dispatch(UpdateEmailAuthOptionsPage(
             mode: EmailAuthMode.signIn,
-            email: '',
-            password: '',
-            repeatPassword: '',
-            autovalidate: AutoValidate.disabled,
```
Comment (tamari-gray) [inline lib/widgets/auth/email_auth_options_page/links/sign_in_link.dart:23]: ran into a problem while manual testing, 
if the user navigates between sign in/sign up screens **without** entering any text, autovalidation is still set to Autovalidate.onUserInput. I know we want those UX points so I suggest we only set autovalidation (after navigation between screens) if the user has entered text

Comment (tamari-gray) [reply lib/widgets/auth/email_auth_options_page/links/sign_in_link.dart:23]: sorry if this wasnt the clearest, im blaming 4pm on a friday 😅. happy to discuss if wsnt clear

**Review by tamari-gray — CHANGES_REQUESTED** [review]
just that problem thing

File: lib/widgets/auth/email_auth_options_page/links/sign_in_link.dart (hunk 1 above)
Comment (nickmeinhold) [reply lib/widgets/auth/email_auth_options_page/links/sign_in_link.dart:23]: super clear, thanks mate!  I'll have a crack at fixing

Comment (nickmeinhold) [reply lib/widgets/auth/email_auth_options_page/links/sign_in_link.dart:23]: I wasn't able to repro on iPhone emulator or a Pixel4, so I'm going to push this through and add an Asana task for Tam to check (some time after Monday) if the problem still exists.  Thanks!

**Pull request merged**